### PR TITLE
Feature/cdap 2444 cloudera docs

### PR DESCRIPTION
**Pull request by dereklwood**
Documentation improvements regarding CSD install/upgrade. This is for CDAP-2444, and will resolve it after the specific 3.1 upgrade is fully tested.

### CODE REVIEW

**Comment by wolf31o2** [comment]
:+1: LGTM, assuming the includes are all correct

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 1]
```diff
@@ -184,7 +141,11 @@ When a new compatible CDAP parcel is released, it will be available via the Parc
 
 .. rubric:: Upgrading Major/Minor Release versions
 
-These steps will upgrade from CDAP 2.8.0 to CDAP 3.0.0. (**Note:** Apps need to be both recompiled and re-deployed.)
+Upgrading between major versions of CDAP involves the additional steps of upgrading the CSD, and running the included
+CDAP Upgrade Tool. Upgrades between multiple Major/Minor versions must be done consecutively, and a version cannot be
+skipped unless otherwise noted.  **Note:** Apps need to be both recompiled and re-deployed.
```
Comment (sreevatsanraman) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: **Note:** Apps need to be both recompiled and re-deployed. 
This is only if we use Timepartitioned file sets.

Comment (sreevatsanraman) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: Note: Apps need to be both recompiled and re-deployed if it uses PartitionedFileSet or TimePartitionedFileSet

**Comment by sreevatsanraman** [comment]
One comment. Fix and merge. LGTM otherwise

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: Note: An app need to be both recompiled and re-deployed if it uses either a PartitionedFileSet or a TimePartitionedFileSet.

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 2]
```diff
@@ -193,24 +154,36 @@ These steps will upgrade from CDAP 2.8.0 to CDAP 3.0.0. (**Note:** Apps need to
...
 
+.. _cloudera-release-specific-upgrade-notes:
+
+.. rubric:: Upgrading CDAP 3.0 to 3.1
+
+No known additional requirements.
+
+.. rubric:: Upgrading CDAP 2.8 to 3.0
+
+When upgrading from 2.8.0 to 3.0.0, the CDAP Web-App role has been replaced by the CDAP-UI role.  After starting the 3.0 services the first time:
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: services for the first time:

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 3]
```diff
@@ -193,24 +154,36 @@ These steps will upgrade from CDAP 2.8.0 to CDAP 3.0.0. (**Note:** Apps need to
...
+
+   - From the CDAP Instances page, select Add Role Instances, and choose a host for the CDAP-UI role.
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: "Add Role Instances"

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 4]
```diff
@@ -193,24 +154,36 @@ These steps will upgrade from CDAP 2.8.0 to CDAP 3.0.0. (**Note:** Apps need to
...
+
+   - From the CDAP Instances page, check the CDAP-Web-App role, and select Delete from the Actions menu.
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: "Delete"

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 5]
```diff
@@ -224,6 +197,18 @@ query, try setting ``hive.exec.stagingdir`` in your Hive configuration to
 This can be done in Cloudera Manager using the *Hive Client
 Advanced Configuration Snippet (Safety Valve) for hive-site.xml* configuration field.
 
+.. rubric:: Missing Application Templates
+
+The bundled application templates are included in the CDAP parcel, located in a subdirectory
+of Cloudera's ``${PARCELS_ROOT}`` directory, for example:
+
+.. parsed-literal::
+  /opt/cloudera/parcels/CDAP/master/templates
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: There's nothing here that needs replacing by the parser; is something missing?

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 6]
```diff
@@ -224,6 +197,18 @@ query, try setting ``hive.exec.stagingdir`` in your Hive configuration to
...
+
+Ensure the ``App Template Dir`` configuration option points to this path on disk. Since this
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: Ensure that the ...

**Comment by awholegunch** [comment]
Minor nits, with one exception: the parsed-literal didn't have anything special: was something left out?

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst (hunk 1 above)
Comment (dereklwood) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: This statement was from the generalized instruction list.  I can put this clarification in the "3.0 to 3.1" release-specific instructions.  Do we not want this as part of general major upgrade instructions?  (ie, it applied to 2.8 -> 3.0 as well)

File: cdap-docs/integrations/source/partners/cloudera/configuring.rst [hunk 7]
```diff
@@ -76,6 +76,7 @@ Download
 =======================================
 
 Download the CDAP CSD (Custom Service Descriptor): `download JAR file <http://cask.co/resources/#cdap-integrations>`__.
```
Comment (awholegunch) [inline cdap-docs/integrations/source/partners/cloudera/configuring.rst]: download the JAR file

**Comment by awholegunch** [comment]
One minor nit, otherwise LGTM. I built this branch, and found a number of errors outside this file in the build, probably due to it requiring a merge from release/3.1. I will do that merge locally to check that it builds correctly.

**Comment by awholegunch** [comment]
Merged branch builds clean.

**Comment by awholegunch** [comment]
LGTM.

**Pull request merged**